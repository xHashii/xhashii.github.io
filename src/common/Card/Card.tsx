import React, { FC, HTMLAttributes } from "react";
import classNames from "classnames";
import { Classes } from "jss";

interface ICardProps extends HTMLAttributes<HTMLDivElement> {
  classes: Classes<"root">;
  className?: string;
  children?: any;
}

const Card: FC<ICardProps> = ({
  classes = {},
  children = null,
  className = "",
  ...props
}) => (
  <div className={classNames(classes.root, className)} {...props}>
    {children}
  </div>
);

export default Card;
