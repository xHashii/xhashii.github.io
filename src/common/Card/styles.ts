export default {
  root: {
    background: "#fff",
    borderRadius: 4,
    display: "inline-block",
    position: "relative" as "relative",
    padding: 32,
    paddingBottom: 24,
    margin: 8,
    zIndex: 0
  }
};
