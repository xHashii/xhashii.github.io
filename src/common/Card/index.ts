import withSyles from "react-jss";

import Card from "./Card";
import styles from "./styles";

export default withSyles(styles)(Card);
