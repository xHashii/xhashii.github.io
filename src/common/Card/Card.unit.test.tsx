import React from "react";
import { shallow } from "enzyme";

import Card from "./Card";

const defaultProps = {
  classes: {}
};

describe("The Card instance", () => {
  describe("When called with default props", () => {
    it("should render without crashing", () => {
      const wrapper = shallow(<Card {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
