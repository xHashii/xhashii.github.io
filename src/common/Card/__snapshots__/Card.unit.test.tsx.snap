// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Card instance When called with default props should render without crashing 1`] = `ShallowWrapper {}`;
